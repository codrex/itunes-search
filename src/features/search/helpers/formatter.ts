import { SearchResponsePayload } from "../../types";

export type FormattedMusicSearch = ReturnType<typeof formatMusicSearch>;

export const formatMusicSearch = (
  data: SearchResponsePayload["results"][number]
) => {
  const {
    artistName,
    artistType,
    kind,
    collectionName,
    collectionType,
    artworkUrl100,
    wrapperType,
    primaryGenreName,
    trackName,
    trackId,
    collectionId,
    artistId,
  } = data;
  switch (wrapperType) {
    case "artist":
      return {
        id: artistId,
        image: `https://ui-avatars.com/api/?size=300&name=${artistName}`,
        type: artistType,
        name: artistName,
        genre: primaryGenreName,
      };
    case "collection":
      return {
        id: collectionId,
        image: artworkUrl100,
        type: collectionType,
        name: collectionName,
        genre: primaryGenreName,
      };
    case "track":
      return {
        id: trackId,
        image: artworkUrl100,
        type: kind,
        name: trackName,
        genre: primaryGenreName,
      };
    default:
      return undefined;
  }
};
