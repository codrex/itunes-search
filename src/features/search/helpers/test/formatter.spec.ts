import { formatMusicSearch } from "../formatter";

describe("formatter", () => {
  it("should format artist data type", () => {
    const payload = {
      artistName: "",
      artistType: "",
      kind: "",
      collectionName: "",
      collectionType: "",
      artworkUrl100: "",
      wrapperType: "artist",
      primaryGenreName: "",
      trackName: "",
      trackId: "",
      collectionId: "",
      artistId: "artist id",
    };
    expect(formatMusicSearch(payload as any)).toMatchSnapshot()
  });

  it("should format collection data type", () => {
    const payload = {
      artistName: "",
      artistType: "",
      kind: "",
      collectionName: "",
      collectionType: "",
      artworkUrl100: "",
      wrapperType: "collection",
      primaryGenreName: "",
      trackName: "",
      trackId: "",
      collectionId: "collection id",
      artistId: "",
    };
    expect(formatMusicSearch(payload as any)).toMatchSnapshot()
  });

  it("should format track data type", () => {
    const payload = {
      artistName: "",
      artistType: "",
      kind: "",
      collectionName: "",
      collectionType: "",
      artworkUrl100: "",
      wrapperType: "track",
      primaryGenreName: "",
      trackName: "",
      trackId: "track id",
      collectionId: "",
      artistId: "",
    };
    expect(formatMusicSearch(payload as any)).toMatchSnapshot()
  });
});
