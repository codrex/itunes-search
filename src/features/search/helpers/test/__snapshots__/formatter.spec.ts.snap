// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatter should format artist data type 1`] = `
Object {
  "genre": "",
  "id": "artist id",
  "image": "https://ui-avatars.com/api/?size=300&name=&background=random",
  "name": "",
  "type": "",
}
`;

exports[`formatter should format collection data type 1`] = `
Object {
  "genre": "",
  "id": "collection id",
  "image": "",
  "name": "",
  "type": "",
}
`;

exports[`formatter should format track data type 1`] = `
Object {
  "genre": "",
  "id": "track id",
  "image": "",
  "name": "",
  "type": "",
}
`;
